src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1448,30 +1446,29 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         auto update{ winrt::make<ScrollPositionChangedArgs>(viewTop,
                                                             viewHeight,
                                                             bufferSize) };
-        if (!_inUnitTests && _updateScrollBar)

```

Comment:
Here's the incorrect if condition. It would trigger as soon as we `_updateScrollBar.reset()`.